src/main/java/com/nooul/apihelpers/springbootrest/specifications/CustomSpecifications.java

```diff
@@ -99,197 +164,206 @@ public Predicate handleAllCases(CriteriaBuilder builder, Root root, Join join, C
-                return builder.equal(root.get(a.getName()), Enum.valueOf(Class.class.cast(a.getJavaType()), (String) val));
-            } else if (isPrimitive(a)) {
-                return builder.equal(root.get(a.getName()), val);
-            } else if(isUUID(a)) {
-                return builder.equal(root.get(a.getName()), UUID.fromString(val.toString()));
-            } else if(a.isAssociation()) {
-                if (isPrimaryKeyOfAttributeUUID(a, root)) {
-                    return prepareJoinAssociatedPredicate(builder, root, a, UUID.fromString(val.toString()));
+        } else if (join == null) {
+            if (isAttributeEnum(attribute)) {
+                return builder.equal(root.get(attribute.getName()),
+                        Enum.valueOf(Class.class.cast(attribute.getJavaType()), (String) val));
+            } else if (isPrimitive(attribute)) {
+                return builder.equal(root.get(attribute.getName()), val);
+            } else if (isAttributeUUID(attribute)) {
+                return builder.equal(root.get(attribute.getName()), UUID.fromString(val.toString()));
+            } else if (attribute.isAssociation()) {
+                if (isPrimaryKeyUUID(attribute, root)) {
+                    return prepareJoinAssociatedPredicate(builder, root, attribute, UUID.fromString(val.toString()));
+                } else {
+                    return prepareJoinAssociatedPredicate(builder, root, attribute, val);
                 }
-                else {
-                    return prepareJoinAssociatedPredicate(builder, root, a, val);
-                }
-            }
-        }
-        else if (join != null) {
-            if (isEnum(a)) {
-                return builder.equal(join.get(a.getName()), Enum.valueOf(Class.class.cast(a.getJavaType()), (String) val));
-            } else if (isPrimitive(a)) {
-                return builder.equal(join.get(a.getName()), val);
-            } else if (a.isAssociation()) {
-                return builder.equal(join.get(a.getName()), val);
             }
-            else if(a.isCollection()) {
+        } else if (join != null) {
+            if (isAttributeEnum(attribute)) {
+                return builder.equal(join.get(attribute.getName()),
+                        Enum.valueOf(Class.class.cast(attribute.getJavaType()), (String) val));
+            } else if (isPrimitive(attribute)) {
+                return builder.equal(join.get(attribute.getName()), val);
+            } else if (attribute.isAssociation()) {
+                return builder.equal(join.get(attribute.getName()), val);
+            } else if (attribute.isCollection()) {
                 return builder.equal(join, val);
             }
         }
         throw new IllegalArgumentException("equality/inequality is currently supported on primitives and enums");
     }
 
-    private Predicate createLikePredicate(CriteriaBuilder builder, Root<T> root, Join join, Attribute a, String val) {
+    /**
```

Comment:
honestly I hate those comments.. that just denote the params :P and say what the method name says - do you use it for javadoc? I think this is implementation detail and there is no need for documentation

Reply:
It's comment autocreation for javadoc. It's perfectly fine to skip this.